### DISCUSSION TITLE
Rolling back feature ?

### DISCUSSION THREAD

**Discussion by Kiddinglife** [UIUC-PPL/charm#3445] [Q&A] [answered]
Does charm support rolling back to the previous time point via time wrap protocol by recomputation of the events ?

**Comment by ericjbohm** [comment, marked as answer]
Hello,

Charm++ is not a Discrete Event Simulation environment, so it does not directly support rollbacks.  However, discrete event simulation has been implemented above Charm++.

See the Charades project for the academic version http://charmplusplus.net/charades_docs/index.html

Or contact Charmworks, Inc. regarding CharmDES, a commercially supported version https://www.hpccharm.com/charmdes
